packages/react-native/src/survey-web-view.tsx

```diff
@@ -205,21 +212,68 @@ export function SurveyWebView({ survey }: SurveyWebViewProps): JSX.Element | und
             if (onDisplay) {
               const { id } = await createDisplay(survey.id);
               surveyState.updateDisplayId(id);
+
+              const existingDisplays = appConfig.get().personState.data.displays;
+              const newDisplay = { surveyId: survey.id, createdAt: new Date() };
+
+              // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- existingDisplays is not always defined
+              const displays = existingDisplays ? [...existingDisplays, newDisplay] : [newDisplay];
```

Comment:
According to typedef of `TJsPersonState`, existingDisplays is a required field and  getPersonState also seems to return an array every time, so can you check if we can remove the check here, along with the comment ?